scipy/special/_round.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Functions for adding two double precision numbers with rounding to
+ * infinity or rounding to negative infinity without using <fenv.h>.
+ *
+ * References
+ * ----------
+ * [1] Hida, Li, Bailey, "Library for Double-Double and Quad-Double
+ *     Arithmetic."
+ */
+#ifndef ROUND_H
+#define ROUND_H
+
+#include <numpy/npy_math.h>
+#include "_c99compat.h"
+
+
+double add_round_up(double a, double b)
+{
+    double s, e, v;
```

Comment:
It may also break on without unsafe optimizations on x87, and I'm not sure if C (or C99) standards specify the precision used in intermediate results.